js/app/modules/arrangement/constrained.js

```diff
@@ -0,0 +1,124 @@
+// Copyright 2016 Endless Mobile, Inc.
+
+/* exported Constrained */
+
+const Emeus = imports.gi.Emeus;
+const Gtk = imports.gi.Gtk;
+const Lang = imports.lang;
+
+const Arrangement = imports.app.interfaces.arrangement;
+const Card = imports.app.interfaces.card;
+const Module = imports.app.interfaces.module;
+
+const _DEFAULT_CARD_COUNT = 10;
+const _DEFAULT_COLUMNS = 5;
+const _DEFAULT_ROWS = 6;
+const _CARD_MIN_WIDTH = Card.MinSize.B;
+const _CARD_MIN_HEIGHT = Card.MinSize.B;
+
+const _Placeholder = new Lang.Class({
+    Name: 'Placeholder',
+    Extends: Gtk.Frame,
+});
+
+/**
+ * Class: Constrained
+ * Arrangement with featured cards on a irregular pattern
+ *
+ * This arrangement shows featured cards that follow a irregular pattern
+ * in three different card sizes. The pattern is described using Emeus
+ * VFL (Visual Format Language).
+ *
+ * This arrangement can be extended by programmers to implement other
+ * constrained arrangements by simply providing a VFL description and
+ * the number of cards.
+ *
+ * To extend this class override the Arrangement.get_max_cards() and
+ * Constrained.get_description().
+ */
+const Constrained = new Module.Class({
+    Name: 'Arrangement.Constrained',
+    Extends: Emeus.ConstraintLayout,
+    Implements: [Arrangement.Arrangement],
+
+    // Arrangement override
+    get_max_cards: function () {
+        return _DEFAULT_CARD_COUNT;
+    },
+
+    // Arrangement override
+    set_models: function (models) {
+        Arrangement.Arrangement.set_models(this, models);
+        this._setup_constraints();
+    },
+
+    // Arrangement override
+    unpack_card: function (card) {
+        let child = card.get_parent();
+        child.remove(card);
+        this.remove(child);
+    },
+
+    /**
+     * Method: get_description
+     * Get Emeus VFL description for this arrangement.
+     */
+    get_description: function () {
+        return [
+            'H:|[view0(view4 * 3)][view1(view4 * 2)]|',
+            'H:|[view0][view2(view4 * 2)]|',
+            'H:|[view3(view4 * 2)][view4][view2]|',
+            'H:|[view3][view5(view4 * 2)][view6(view4)]|',
+            'H:|[view8(view4)][view9(view4)][view5][view7(view4)]|',
+            'V:|[view0(view4 * 3)][view3(view4 * 2)][view8(view4)]|',
+            'V:[view3][view9(view4)]|',
+            'V:|[view0][view4][view5(view4 * 2)]|',
+            'V:|[view1(view4 * 2)][view2(view4 * 2)][view6(view4)][view7(view4)]|',
+        ];
+    },
+
+    _setup_constraints: function () {
```

Comment:
As we agreed on the hangout call: a) I can't leave it as a property because we don't want to expose the VFL yet, and b) that override of set_models hides implementation details, so we don't want other classes to re-implement it.